### PR TITLE
custom advertised prefix for primary vxlan tunnel [202012] 

### PR DESCRIPTION
**Pull request by siqbal1986**
**What I did**
Added support to advertise a different prefix rather than a route prefix. This commit also changes the behavior to create NH group based on primary endpoints. This change doesnt support switchover to endpoint monitoring( custom or otherwise.)

**Why I did it**
To cater to a use case where a common prefix range can be advertised by multiple routes to avoid BGP route table overload.

**How I verified it**
Added IPV4 and IPV6 tests

**Details if related**
Pushing to 202012 first for urgency.

### CODE REVIEW

File: tests/test_vnet.py [hunk 1]
```diff
@@ -2424,36 +2424,201 @@ def test_vnet_orch_17(self, dvs, testlog):
...
+        vnet_obj.check_vnet_entry(dvs, vnet_name)
+        vnet_obj.check_vxlan_tunnel_entry(dvs, tunnel_name, vnet_name, '10009')
 
         vnet_obj.check_vxlan_tunnel(dvs, tunnel_name, '9.9.9.9')
 
         vnet_obj.fetch_exist_entries(dvs)
-        create_vnet_routes(dvs, "100.100.1.1/32", 'Vnet18', '9.0.0.1,9.0.0.2,9.0.0.3', ep_monitor='9.1.0.1,9.1.0.2,9.1.0.3',primary ='9.0.0.1',monitoring='custom', adv_prefix='100.100.1.1/27')
+ 
+        #Add first ROute
+        create_vnet_routes(dvs, "100.100.1.1/32", vnet_name, '9.0.0.1,9.0.0.2,9.0.0.3', ep_monitor='9.1.0.1,9.1.0.2,9.1.0.3', profile = "test_prf", primary ='9.0.0.1',monitoring='custom', adv_prefix='100.100.1.1/24')
```
Comment (prsunny) [inline tests/test_vnet.py:2447]: adv_prefix must be 100.100.1.0/24

File: tests/test_vnet.py [hunk 2]
```diff
@@ -2424,36 +2424,201 @@ def test_vnet_orch_17(self, dvs, testlog):
...
+
+        #remove first route
+        delete_vnet_routes(dvs, "100.100.1.1/32", vnet_name)
+        vnet_obj.check_del_vnet_routes(dvs, 'Vnet12', ["100.100.1.1/32"])
+        check_remove_state_db_routes(dvs, 'Vnet12', "100.100.1.1/32")
+
+        #adv should be gone.
+        check_remove_routes_advertisement(dvs, "100.100.1.1/24")
+
+        #add 2nd route
```
Comment (prsunny) [inline tests/test_vnet.py:2599]: please add a test for changing primary without changing nexthops

File: tests/test_vnet.py [hunk 3]
```diff
@@ -2424,36 +2424,201 @@ def test_vnet_orch_17(self, dvs, testlog):
...
+        create_vnet_entry(dvs, vnet_name, tunnel_name, '10009', "", advertise_prefix=True, overlay_dmac="22:33:33:44:44:66")
+
+        vnet_obj.check_vnet_entry(dvs, vnet_name)
+        vnet_obj.check_vxlan_tunnel_entry(dvs, tunnel_name, vnet_name, '10009')
+
+        vnet_obj.check_vxlan_tunnel(dvs, tunnel_name, '9.9.9.9')
+
+        vnet_obj.fetch_exist_entries(dvs)
+        import pdb
+        pdb.set_trace()
```
Comment (prsunny) [inline tests/test_vnet.py:2576]: Please remove this

File: tests/test_vnet.py [hunk 4]
```diff
@@ -2424,36 +2424,201 @@ def test_vnet_orch_17(self, dvs, testlog):
...
+
+        #add 2nd route
+        create_vnet_routes(dvs, "100.100.1.57/32", vnet_name, '5.0.0.1,5.0.0.2,5.0.0.3,5.0.0.4', ep_monitor='5.1.0.1,5.1.0.2,5.1.0.3,5.1.0.4', profile = "test_prf", primary ='5.0.0.1,5.0.0.2',monitoring='custom', adv_prefix='100.100.1.1/24')
+        route2 , _ = vnet_obj.check_vnet_ecmp_routes(dvs, vnet_name, ['5.0.0.1','5.0.0.2'], tunnel_name)
+        check_state_db_routes(dvs, vnet_name, "100.100.1.57/32", ['5.0.0.1,5.0.0.2'])
+        # The default Vnet setting does not advertise prefix
+        check_routes_advertisement(dvs, "100.100.1.1/24")
+
+ 
+         #modify  2nd route next hops
```
Comment (prsunny) [inline tests/test_vnet.py:2607]: please fix alignment

File: tests/test_vnet.py [hunk 5]
```diff
@@ -2424,36 +2424,201 @@ def test_vnet_orch_17(self, dvs, testlog):
...
+        #remove 3rd route
+        delete_vnet_routes(dvs, "100.100.1.64/32", vnet_name)
+        vnet_obj.check_del_vnet_routes(dvs, 'Vnet12', ["100.100.1.64/32"])
+        check_remove_state_db_routes(dvs, 'Vnet12', "100.100.1.64/32")
+
+        #adv should be gone.
+        check_remove_routes_advertisement(dvs, "100.100.1.1/24")
+
+    '''
+    Test 15 - Test for vxlan custom monitoring with adv_prefix-no monitoring IPV6. temporary
```
Comment (prsunny) [inline tests/test_vnet.py:2492]: Test 19

File: tests/test_vnet.py (hunk 1 above)
Comment (siqbal1986) [reply tests/test_vnet.py:2447]: done

File: tests/test_vnet.py (hunk 2 above)
Comment (siqbal1986) [reply tests/test_vnet.py:2597]: done

File: tests/test_vnet.py (hunk 3 above)
Comment (siqbal1986) [reply tests/test_vnet.py]: done

File: tests/test_vnet.py (hunk 5 above)
Comment (siqbal1986) [reply tests/test_vnet.py]: done

**Review by prsunny — APPROVED** [review]

**Pull request merged**